src/host/selectionInput.cpp

```diff
@@ -390,20 +390,30 @@ bool Selection::HandleKeyboardLineSelectionEvent(const INPUT_KEY_INFO* const pIn
             // shift + pgup/pgdn extends selection up or down one full screen
         case VK_NEXT:
         {
-            coordSelPoint.Y += sWindowHeight; // TODO: potential overflow
-            if (coordSelPoint.Y > bufferSize.BottomInclusive())
+            // cast up to int and check to avoid overflow
+            const int newY = coordSelPoint.Y + sWindowHeight;
+            if (newY > bufferSize.BottomInclusive())
             {
                 coordSelPoint.Y = bufferSize.BottomInclusive();
             }
+            else
+            {
+                coordSelPoint.Y = static_cast<SHORT>(newY);
+            }

```

Comment:
I'd sort of hope for this fix to be something like 
`coordSelPoint.Y = base::CheckedAdd(coordSelPoint.Y, sWindowHeight).ValueOrDefault(bufferSize.BottomInclusive());` as we have the nice checked math libraries to handle things like this now (and we probably didn't when this code was written).

Reply:
I guess, (though that does add a bunch of unneeded complexity, as now we are checking for the value to be in range AND then greater than bufferSize.BottomInclusive(), whereas any value greater than SHORT_MAX is already greater than bufferSize.BottomInclusive()).

Reply:
There's a lot of value in having similar algorithmic checks go through the same sort of guard function, in my mind. Even without adding anything in particular value in terms of the correctness, using the library function marks this as a point where error could occur.

Also, if you want to shorten the latter part... you could theoretically use std::min as well for the smaller of the checked number and bottom inclusive.

Reply:
Thanks, but a simple comparison seems more appropriate for the latter part.

Reply:
Anyway, I changed the check to use base:: now